ports/stm32f4/common-hal/busio/UART.c

```diff
@@ -36,45 +36,455 @@
+    #ifdef UART4
+        reserved_uart[3] = false;
+        MP_STATE_PORT(cpy_uart_obj_all)[3] = NULL;
+        __HAL_RCC_UART4_FORCE_RESET();
+        __HAL_RCC_UART4_RELEASE_RESET();
+        __HAL_RCC_UART4_CLK_DISABLE();
+    #endif
+    #ifdef UART5
+        reserved_uart[4] = false;
+        MP_STATE_PORT(cpy_uart_obj_all)[4] = NULL;
+        __HAL_RCC_UART5_FORCE_RESET();
+        __HAL_RCC_UART5_RELEASE_RESET();
+        __HAL_RCC_UART5_CLK_DISABLE();
+    #endif
+    #ifdef USART6
+        reserved_uart[5] = false;
+        MP_STATE_PORT(cpy_uart_obj_all)[5] = NULL;
+        __HAL_RCC_USART6_FORCE_RESET();
+        __HAL_RCC_USART6_RELEASE_RESET();
+        __HAL_RCC_USART6_CLK_DISABLE(); 
+    #endif
+}
+
+STATIC USART_TypeDef * assign_uart_or_throw(busio_uart_obj_t *self, bool pin_eval, 
+                                            int uart_index, bool uart_taken) {
+    if(pin_eval) {
+        //assign a root pointer pointer for IRQ
+        MP_STATE_PORT(cpy_uart_obj_all)[uart_index] = self;
+        return mcu_uart_banks[uart_index];
+    } else {
+        if (uart_taken) {
+            mp_raise_ValueError(translate("Hardware busy, try alternative pins"));
+        } else {
+            mp_raise_ValueError(translate("Invalid UART pin selection"));
+        }
+    }
+}
+
+
+STATIC void uart_clk_irq_enable(busio_uart_obj_t *self, USART_TypeDef * USARTx) {
+    #ifdef USART1
+    if(USARTx==USART1) { 
+        reserved_uart[0] = true;
+        __HAL_RCC_USART1_FORCE_RESET();
+        __HAL_RCC_USART1_RELEASE_RESET();
+        __HAL_RCC_USART1_CLK_ENABLE();
+        self->irq = USART1_IRQn;
+    } 
+    #endif
+    #ifdef USART2
+    if(USARTx==USART2) { 
+        reserved_uart[1] = true;
+        __HAL_RCC_USART2_FORCE_RESET();
+        __HAL_RCC_USART2_RELEASE_RESET();
+        __HAL_RCC_USART2_CLK_ENABLE();
+        self->irq = USART2_IRQn;
+    } 
+    #endif
+    #ifdef USART3
+    if(USARTx==USART3) { 
+        reserved_uart[2] = true;
+        __HAL_RCC_USART3_FORCE_RESET();
+        __HAL_RCC_USART3_RELEASE_RESET();
+        __HAL_RCC_USART3_CLK_ENABLE();
+        self->irq = USART3_IRQn;
+    } 
+    #endif
+    #ifdef UART4
+    if(USARTx==UART4) { 
+        reserved_uart[3] = true;
+        __HAL_RCC_UART4_FORCE_RESET();
+        __HAL_RCC_UART4_RELEASE_RESET();
+        __HAL_RCC_UART4_CLK_ENABLE();
+        self->irq = UART4_IRQn;
+    } 
+    #endif
+    #ifdef UART5
+    if(USARTx==UART5) { 
+        reserved_uart[4] = true;
+        __HAL_RCC_UART5_FORCE_RESET();
+        __HAL_RCC_UART5_RELEASE_RESET();
+        __HAL_RCC_UART5_CLK_ENABLE();
+        self->irq = UART5_IRQn;
+    } 
+    #endif
+    #ifdef USART6
+    if(USARTx==USART6) { 
+        reserved_uart[5] = true;
+        __HAL_RCC_USART6_FORCE_RESET();
+        __HAL_RCC_USART6_RELEASE_RESET();
+        __HAL_RCC_USART6_CLK_ENABLE();
+        self->irq = USART6_IRQn;
+    } 
+    #endif
+}
 
 void common_hal_busio_uart_construct(busio_uart_obj_t *self,
```

Comment:
Could you clean up the syntax style in this routine: space after `for`; space after `;`, spaces around operators, etc.? Thanks.